src/tests/zfs/zfs_custom_node_topology/zfs_custom_node_topology_test.go

```diff
@@ -0,0 +1,397 @@
+	time.Sleep(10 * time.Minute)
+
+	productConfig := e2e_config.GetConfig().Product
+	label := fmt.Sprintf("%s=%s", productConfig.LocalEngineComponentPodLabelKey,
+		productConfig.ZfsEngineComponentDsPodLabelValue)
+
+	// restart zfs daemonset pods
+	err = k8stest.DeletePodsByLabel(label, common.NSOpenEBS())
+	Expect(err).To(BeNil(), "failed to restart zfs daemonset pods with label %s", label)
+
+	// verify zfs daemonset to be ready
+	Eventually(func() bool {
+		return k8stest.DaemonSetReady(productConfig.ZfsEngineDaemonSetName, common.NSOpenEBS())
+	},
+		defDaemonsetReadyTime,
+		"5s",
+	).Should(BeTrue())
+
+	ready, err := k8stest.OpenEBSReady(10, 540)
+	Expect(err).To(BeNil(), "failed to verify openebs pods running state")
+	Expect(ready).To(BeTrue(), "some of the openebs pods are not running")
+
+	// verify topology key in csi node
+	var csiNodeErr error
+	Eventually(func() bool {
+		var isKeyFound bool
+		isKeyFound, csiNodeErr = k8stest.CheckCsiNodeTopologyKeysPresent(targetNode,
+			productConfig.ZfsEnginePluginDriverName,
+			[]string{
+				key,
+			})
+		return isKeyFound
+	},
+		csiNodeUpdateTime,
+		"5s",
+	).Should(BeTrue())
+
+	Expect(csiNodeErr).ToNot(HaveOccurred(), "failed to get csi node %s, %v", targetNode, err)
+
+	// deploy fio pods for created zfs volumes
+	for ix, app := range appInstances {
+		logf.Log.Info(fmt.Sprintf("%d)", ix), "fio-pod", app.Decor)
+
+		// deploy fio pod with created volume
+		logf.Log.Info("deploy fio pod with created volume")
+		err = app.DeployApplication()
+		Expect(err).To(BeNil(), "failed to deploy app")
+
+		//verify pvc and pv to be bound
+		volUuid, err := k8stest.VerifyVolumeProvision(app.GetPvcName(), common.NSDefault)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(volUuid).ToNot(BeEmpty())
+
+		// use created PVC which is deployed as part of restore app
+		err = app.RefreshVolumeState()
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	// verify all fio application are deployed on same labeled node
+	for ix, app := range appInstances {
+		logf.Log.Info(fmt.Sprintf("%d)", ix), "fio-pod", app.Decor)
+		appPodName := app.GetPodName()
+		logf.Log.Info("app pod", "fio-pod", appPodName)
+		//get node name where where app is deployed
+		node, err := k8stest.GetNodeForPodByPrefix(appPodName, common.NSDefault)
+		Expect(err).ToNot(HaveOccurred(), "failed to node for app pod %s, %v", appPodName, err)
+		Expect(node).Should(Equal(targetNode), "app pod %s does not scheduled on node %s", targetNode)
+	}
+
+	// remove all fio application
+	for ix, app := range appInstances {
+		logf.Log.Info(fmt.Sprintf("%d)", ix), "fio-pod", app.Decor, "err", err)
+		// remove app pod, pvc,sc
+		err = app.Cleanup()
+		Expect(err).To(BeNil(), "failed to clean resources")
+	}
+
+	// Remove the labels from nodes after the end of test
+	err = k8stest.UnlabelNode(targetNode, key)
+	Expect(err).ToNot(HaveOccurred(), "failed to remove label from node %s", targetNode)
+	targetNode = ""
+
+	// restart zfs daemonset pods
```

Comment:
Same here

Reply:
topology key present in csinode kubernetes object for local.csi.openebs.io plugin driver should be removed before starting new topology test and to do so , daemonset pods needs to restated after removing node label with the key